src/screenshots/crop.js

```diff
@@ -0,0 +1,80 @@
+import fs from 'fs';
+import Promise from 'pinkie';
+import { PNG } from 'pngjs';
+import promisifyEvent from 'promisify-event';
+import { InvalidElementScreenshotDimensionsError } from '../errors/test-run/';
+
+
+function readPng (filePath) {
+    var png           = new PNG();
+    var parsedPromise = Promise.race([
+        promisifyEvent(png, 'parsed'),
+        promisifyEvent(png, 'error')
+    ]);
+
+    fs.createReadStream(filePath).pipe(png);
+
+    return parsedPromise
+        .then(() => png);
+}
+
+function writePng (filePath, png) {
+    var outStream     = fs.createWriteStream(filePath);
+    var finishPromise = Promise.race([
+        promisifyEvent(outStream, 'finish'),
+        promisifyEvent(outStream, 'error')
+    ]);
+
+    png.pack().pipe(outStream);
+
+    return finishPromise;
+}
+
+function enforceDimension (value, min, max) {
```

Comment:
may be move this function to `src/utils/` and call e.g. `enforceInRange` to get rid of two [similar functions](https://github.com/DevExpress/testcafe/pull/2026/files#diff-6f3455b3adb657ef8e557804aba41f0aR4)